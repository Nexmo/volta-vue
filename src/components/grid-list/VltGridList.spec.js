import { mount } from '@vue/test-utils';
import { VltGridList } from '..';

const columns = [
  { title: 'Name', sortable: true, property: 'name' },
  { title: 'Position', property: 'position' },
];

const rows = [
  { name: 'Louis', position: 'Front End Developer' },
  { name: 'Mie', position: 'UX Designer' },
  { name: 'Julien', position: 'Java Developer' },
  { name: 'Nisha', position: 'JavaScript Developer' },
];

const fullGridProps = {
  columns,
  rows,
};

describe('vlt-grid-list', () => {
  let wrapper;

  test('renders empty state', () => {
    wrapper = mount(VltGridList);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders correctly with two columns, "Name" and "Position"', () => {
    wrapper = mount(VltGridList, { propsData: fullGridProps });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('pagination is shown', () => {
    wrapper = mount(VltGridList, {
      propsData: {
        ...fullGridProps,
        pageSize: 1,
        pagination: true,
        total: 4,
      },
    });
    expect(wrapper.find('.Vlt-table__pagination')).toMatchSnapshot();
  });

  test('"page-change" is emitted with "3" when navigating to page "3"', () => {
    wrapper = mount(VltGridList, {
      propsData: {
        ...fullGridProps,
        pageSize: 1,
        pagination: true,
        total: 4,
      },
    });
    wrapper.findAll('.Vlt-table__pagination a').at(2).trigger('click');
    expect(wrapper.emitted('page-change')).toEqual([[3]]);
  });
});
