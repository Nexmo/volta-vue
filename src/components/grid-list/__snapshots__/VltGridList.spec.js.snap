// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-grid-list pagination is shown 1`] = `
<div class="Vlt-table__pagination">
  <ul>
    <!---->
    <li class="Vlt-table__pagination__current"><a href="#">1</a></li>
    <!---->
    <li class=""><a href="#">2</a></li>
    <li class=""><a href="#">3</a></li>
    <!---->
    <li class=""><a href="#">4</a></li>
    <li><a href="#">Next</a></li>
  </ul>
</div>
`;

exports[`vlt-grid-list renders correctly with two columns, "Name" and "Position" 1`] = `
<div
  columns="[object Object],[object Object]"
>
  <div
    class=" Vlt-grid"
    id=""
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`vlt-grid-list renders empty state 1`] = `
<div>
  <!---->
   
  <!---->
   
  <div
    class="Vlt-empty Vlt-empty--search"
  >
    <div
      class="Vlt-empty__content"
    >
      <p>
        There are no results for this search
      </p>
    </div>
  </div>
</div>
`;
